.github/workflows/publish-docs.yml

```
@@ -62,19 +63,56 @@ jobs:
         with:
           node-version: ${{ env.NODE_VERSION }}
 
-      - name: mike deploy main
+      # This deploys the current docs into gh-pages/head on merges to main
+      # The old "main" gets deleted if it exists, head is more descriptive
+      - name: mike deploy head
         if: contains(github.ref, 'refs/heads/main')
         run: |
-          mike deploy --push main
+          mike delete main || true
+          mike deploy --push head
 
-      - name: Get the release version
-        if: contains(github.ref, 'refs/tags/v') && !github.event.release.prerelease # (don't generate for pre-releases)
-        id: get_version
-        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+      # Lists tags using the "gh" cli
+      # Gets rid of old versions (ones without +k0s) and drafts using grep
+      # The list is sorted using k0s_sort installed "install dependencies"
+      - name: Collect tags
+        id: tags
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          TAGS=$(gh release list -L 1000 -R k0sproject/k0s | grep "+k0s." | grep -v Draft | cut -f 1 | k0s_sort)
+          echo ::set-output name=SORTED_TAGS::${TAGS}
```

I think there will be line break issues when trying to pass on the tags without qutes. What about extracting latest and stable directly here, and use them later on via an `env:` block?

Moreover, `k0s_sort` will strip the leading "v", so we could add it back here, as well.

```suggestion
          LATEST=$(echo "$TAGS" | tail -1)
          STABLE=$(echo "$TAGS" | grep -v -- "-" | tail -1)
          echo ::set-output name=LATEST::"v${LATEST}"
          echo ::set-output name=STABLE::"v${STABLE}"
```

---

Hmm, k0s_sort needs to not strip it, I'll fix that there.

---

Combined the steps so the list doesn't need to go to an output